lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -326,6 +359,84 @@ static bool sinkCode(Function *F) {
       }
     }
 
+    // Sink Transpose below ChannelShuffle (Reshape->Transpose->Reshape)
+    if (auto *RN1 = dyn_cast<ReshapeNode>(node)) {
```

Comment:
Can you extract this to its own function? It's a bit long IMO to include along with all the others.